server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/GroupsRoutes.java

```diff
@@ -0,0 +1,187 @@
+import org.apache.james.webadmin.Constants;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSortedSet;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+@Api(tags = "Address Groups")
+@Path(GroupsRoutes.ROOT_PATH)
+@Produces(Constants.JSON_CONTENT_TYPE)
+public class GroupsRoutes implements Routes {
+
+    @VisibleForTesting static final String ROOT_PATH = "address/groups";
+    private static final String GROUP_ADDRESS = "groupAddress";
+    private static final String GROUP_ADDRESS_PATH = ROOT_PATH + SEPARATOR + ":" + GROUP_ADDRESS;
+    private static final String USER_ADDRESS = "userAddress";
+    private static final String USER_IN_GROUP_ADDRESS_PATH = GROUP_ADDRESS_PATH + SEPARATOR + ":" + USER_ADDRESS;
+
+    private final JsonTransformer jsonTransformer;
+    private final RecipientRewriteTable recipientRewriteTable;
+
+    @Inject
+    @VisibleForTesting
+    GroupsRoutes(RecipientRewriteTable recipientRewriteTable, JsonTransformer jsonTransformer) {
+        this.jsonTransformer = jsonTransformer;
+        this.recipientRewriteTable = recipientRewriteTable;
+    }
+
+    @Override
+    public void define(Service service) {
+        service.get(ROOT_PATH, this::listGroups, jsonTransformer);
+        service.get(GROUP_ADDRESS_PATH, this::listGroupMembers, jsonTransformer);
+        service.put(GROUP_ADDRESS_PATH, (request, response) -> halt(HttpStatus.BAD_REQUEST_400));
+        service.put(USER_IN_GROUP_ADDRESS_PATH, this::addToGroup);
+        service.delete(GROUP_ADDRESS_PATH, (request, response) -> halt(HttpStatus.BAD_REQUEST_400));
+        service.delete(USER_IN_GROUP_ADDRESS_PATH, this::removeFromGroup);
+    }
+
+    @GET
+    @Path(ROOT_PATH)
+    @ApiOperation(value = "getting groups list with aphabetical ordering")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "OK", response = List.class),
+        @ApiResponse(code = 500, message = "Internal server error - Something went bad on the server side.")
+    })
+    private Set<String> listGroups(Request request, Response response) throws RecipientRewriteTableException {
+        return Optional.ofNullable(recipientRewriteTable.getAllMappings())
+            .map(mappings ->
+                mappings.entrySet().stream()
+                    .filter(e -> e.getValue().contains(Mapping.Type.Address))
+                    .map(Map.Entry::getKey)
+                    .collect(Guavate.toImmutableSortedSet()))
```

Comment:
no need to explicitly sort on alphabetical ordering?

Reply:
String + naturalOrdering looks fine to me

Reply:
then you mean lexicographic ordering I guess

Reply:
it's not obvious to me what the difference is. Anyway, I should not have wrote that, it's your fault, because ordering is related to l10n and we don't have any clue how to do that for now.

Reply:
in lexicographic, Z < a
anyway I don't care, I just want mappings to be returned in a consistent order. So just remove "with alphabetic order" and I'm happy

Reply:
ok